kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/engine/DefaultJsonHandlerSuite.scala

```diff
@@ -476,4 +481,32 @@ class DefaultJsonHandlerSuite extends AnyFunSuite with TestUtils with DefaultVec
 
     checkAnswer(Seq(actResult), Seq(expResult))
   }
+
+  // See issue https://github.com/delta-io/delta/issues/3888 for original problem
+  test("E2E operationParameters parse map valuesa as strings") {
```

Comment:
maybe we just want to check that this **doesn't break parsing** but let's include an E2E test that uses `tableImpl.getChanges`